### PR TITLE
Add Screenshot functionality to `browse_website`

### PR DESCRIPTION
**Pull request by horazius**
### Background
Added the screenshot options during a browser call I suggested here https://github.com/Significant-Gravitas/Auto-GPT/issues/2443

### Changes
Added the functionality to make a screenshot on every browser call.

### Documentation
added a own function, added a call after the url call in the browsing function.

### Test Plan
Tested it with chrome inside my DEV Container many times for different websites, withour any error.

### PR Quality Checklist
- [x ] My pull request is atomic and focuses on a single change.
- [x ] I have thoroughly tested my changes with multiple different prompts.
- [x ] I have considered potential risks and mitigations for my changes.
- [x ] I have documented my changes clearly and comprehensively.

### CODE REVIEW

**Comment by Tonylib** [comment]
Could you make this a general command that allows screenshot of the full desktop or an option to do so?
The use-case is to allow autoGPT to use the PC (and any application installed) as a tool to accomplish its goals.

I have not opened this PR yet but here is what I'm working on.

1) We need the screenshot command to save a screenshot to a file. ex: screenshot.png
2) This file is given to Meta SAM for segmentation into icons, buttons, text, etc. The output from SAM has everything we need.
3) Ask chatGPT: "Given the following segmentations of a PC desktop where should I click to accomplish the {goal/task}. Reply in json format with the X,Y coordinates of where to double-click or the text to enter via the keyboard"
4) Now we need mouse and keyboard commands. Use the mouse command to move the mouse to X,Y and double-click.
5) Use the keyboard command to enter the Text.
5) REPEAT - But we need some prompt engineering on 3 to make a general purpose prompt that returns the mouse X,Y or Text to be entered.

Once these commands are available I expect autoGPT to use the screenshot, mouse, and keyboard commands to interact with the desktop and use every app available to accomplish the goals.

ChatGPT already knows how to use every application there is - we just need to give it access to the desktop, mouse, and keyboard. AutoGPT is capable of this now with the addition of a few commands and a 1 prompt.

Segmenting the desktop screenshot will produce app icon locations with a bounding box. If an application is open on the desktop the SAM output would naturally include a description of that content as well.

Given the looping nature of autoGPT it should continue using the PC tools to accomplish goals.

**Comment by Reno-Codes** [comment]
> 

I had the same idea, but what if it could be made with OpenCV for desktop object detection in real-time, without the use of Meta SAM.
And along with that, we could make it use PyAutoGUI. It let Python scripts control the mouse and keyboard to automate interactions with other applications. 
GPT-4 is able to write reliable code for interacting with PyAutoGUI and OpenCV

**Comment by Tonylib** [comment]
> openCV for desktop object detection in real-time

I don't think it matters where the source of 'desktop info' comes from - anything that can produce a description of the screen in json format I would expect to work. As long as it contains a list of icons, apps, menus, text and their coordinates or bounding boxes. Any decent LLM should be able to take this json + prompt and produce a json containing our list of keyboard/mouse actions.

I was looking into screen readers like [NVDA](https://www.nvaccess.org/files/nvda/documentation/userGuide.html?#GeneralFeatures)

**Comment by p-i-** [comment]
This is a mass message from the AutoGPT core team.
Our apologies for the ongoing delay in processing PRs.
This is because we are re-architecting the AutoGPT core!

For more details (and for infor on joining our Discord), please refer to:
  https://github.com/Significant-Gravitas/Auto-GPT/wiki/Architecting

**Comment by Pwuts** [comment]
This could be very useful for debugging purposes

File: autogpt/commands/web_selenium.py [hunk 1]
```diff
@@ -206,6 +214,29 @@ def add_header(driver: WebDriver) -> None:
...
+    scroll_width = driver.execute_script("return document.body.scrollWidth;")
+    scroll_height = driver.execute_script("return document.body.scrollHeight;")
+    
+    # Resize the browser window to the dimensions of the web page
+    driver.set_window_size(scroll_width, scroll_height)
+
+    workspace_path = "auto_gpt_workspace"
+    os.makedirs(workspace_path, exist_ok=True)
+    screenshot_path = os.path.join(workspace_path, filename)
+    driver.save_screenshot(screenshot_path)
```
Comment (Pwuts) [inline autogpt/commands/web_selenium.py:237]: This needs to be updated to use `agent.workspace.get_path()`

File: autogpt/commands/web_selenium.py [hunk 2]
```diff
@@ -70,6 +73,11 @@ def browse_website(url: str, question: str, agent: Agent) -> str:
         msg = e.msg.split("\n")[0]
         return f"Error: {msg}"
 
+    #Make a Screenshot
+    timestamp = datetime.now().strftime("%Y%m%d-%H%M%S")
+    screenshot_filename = f"screenshot-{timestamp}.png"
+    save_screenshot(driver, screenshot_filename)
```
Comment (Pwuts) [inline autogpt/commands/web_selenium.py:79]: Does this work in headless mode?

**Review by Pwuts — CHANGES_REQUESTED** [review]

**Comment by LHamnett** [comment]
Commenting for personal reference, may pick up this task and work to resolve this PR when I have time.

**Pull request closed without merging**